src/Microsoft.Health.Dicom.Core/Features/Store/DicomDatasetValidator.cs

```diff
@@ -111,8 +111,20 @@ private void ValidateTags(DicomDataset dicomDataset, IEnumerable<QueryTag> query
             {
                 DicomElement dicomElement = dicomDataset.GetDicomItem<DicomElement>(queryTag.Tag);
 
-                if (dicomElement != null && dicomElement.ValueRepresentation == queryTag.VR)
+                if (dicomElement != null)
                 {
+                    if (dicomElement.ValueRepresentation != queryTag.VR)
+                    {
+                        throw new DatasetValidationException(
+                           FailureReasonCodes.ValidationFailure,
+                           string.Format(
+                               CultureInfo.InvariantCulture,
+                               DicomCoreResource.MismatchVR,
+                               dicomElement.ValueRepresentation,
+                               queryTag.Tag,
```

Comment:
>Tag [](start = 40, length = 3)

Does this consistently represent tag path when path is used and tag name when name is used?

Reply:
this validation happens when customer store Dicom file, and so we don't know the original tag path or tag name.